clients/scheduler/src/utils/dates/functions.ts

```diff
@@ -1,36 +1,50 @@
-export const isTodayOrBefore = (date: Date): boolean => {
-  const today = new Date()
+export const isEarliestDateOrBefore = (
+  date: Date,
+  startDate: Date | null
+): boolean => {
+  const earliestDate = startDate || new Date()
 
   return (
-    date.getDate() <= today.getDate() &&
-    date.getMonth() <= today.getMonth() &&
-    date.getFullYear() <= today.getFullYear()
+    date.getFullYear() < earliestDate.getFullYear() ||
```

Comment:
was getting a cheeky lil error with the prior v of this where something like July 27th, 2024 was considered later than August 8th, 2024 because the 27 in July was > than the 8 in August